src/main/java/org/apache/sling/feature/launcher/impl/Main.java

```diff
@@ -264,20 +270,23 @@ protected static void parseArgs(final LauncherConfig config, final String[] args
                 config.getConfigClashOverrides().put(name, cfgCProps.getProperty(name));
             }
 
-            Properties fwProps = cl.getOptionProperties(OPT_FRAMEWORK_PROPERTIES);
-            for (final String name : fwProps.stringPropertyNames()) {
-                config.getInstallation()
-                        .getFrameworkProperties()
-                        .put(name, fwProps.getProperty(name));
-            }
+            extractValuesFromOption(cl, OPT_FRAMEWORK_PROPERTIES).orElseGet(ArrayList::new)
+                    .forEach(value -> {
+                        final String[] keyVal = split(value);
+                        config.getInstallation().getFrameworkProperties().put(keyVal[0], keyVal[1]);
+                    });
 
             Properties varProps = cl.getOptionProperties(OPT_VARIABLE_VALUES);
             for (final String name : varProps.stringPropertyNames()) {
                 config.getVariables().put(name, varProps.getProperty(name));
             }
 
-            extractValueFromOption(cl, OPT_VERBOSE, "debug").ifPresent(
-                    value -> System.setProperty("org.slf4j.simpleLogger.defaultLogLevel", value));
+            if (cl.hasOption(OPT_VERBOSE)) {
+                extractValueFromOption(cl, OPT_VERBOSE, "debug").ifPresent(value -> System
```

Comment:
This should validate that the supplied verbose option is a valid log level.  It must be one of ("trace", "debug", "info", "warn", "error" or "off").  Any other value should report an error and exit.

Reply:
Sure.

Reply:
It's better.  How would you feel about maybe throwing a ParseException when the logLevel is invalid so it fails fast and does the printHelp call?

Also, it would be good to say what the valid values are in the printHelp output for the verbose option